src/httpclient.h

```diff
@@ -6,18 +6,31 @@
 
 #include "config.h"
 
+struct HttpResponse {
+  HttpResponse(const std::string &body_in, unsigned int http_status_code_in, CURLcode curl_code_in,
+               const std::string &error_message_in)
+      : body(body_in),
+        http_status_code(http_status_code_in),
+        curl_code(curl_code_in),
+        error_message(error_message_in) {}
+  std::string body;
+  unsigned int http_status_code;
+  CURLcode curl_code;
+  std::string error_message;
+  operator bool() const { return curl_code == CURLE_OK; }
+  bool operator!() const { return curl_code != CURLE_OK; }
```

Comment:
What about merging 3 checks together here:

1) That the http request completed OK
2) That the status code was 200 (or 20x if a response isn't needed)
3) That the response is valid JSON?

The JSON parsing could then be moved into HttpResponse, and "Utils::parseJSON(http.get(server + path).body);" would become:

~~~
HttpReponse http_response = http.get(path)
if (http_response.Ok()) {
  Json::Value json = http_response.json();
} else {
  // Handle error
}
~~~